### DISCUSSION TITLE
Having issues connecting to GCS via cyberduck

### DISCUSSION THREAD

**Discussion by NehaShipt** [drakkan/sftpgo#926] [Q&A]
Hi drakkan,

I am trying to connect to my gcp bucket from Cyberduck through a user created on sftp/go.
Steps followed are:

1. Provisioned a GKE cluster with a service account and required permissions.
2. Created a bucket (not sure if i have to give that service account access to this bucket as well - for now it is not there)
3. Deployed sftp/go docker image on my project's container registry and was able to deploy and expose it on the container.
4. Now the sftp/go UI came up and i was able to create an admin user and then a sftpgo user from admin portal.
5. Choose the Google cloud storage as backend and checked the automatic credentials box.
6. Also added my public key for this user.
7. Now when i go to cyberduck and fill in the ip , 2022 as port , username and select my private key file - it is not able to make a connection to the server and upload to bucket.

Some points:
i have not provisioned external DB - doing this for learning purpose so using inbuilt sqlite in sftpgo
Running on 1 pod only to retain data
Do i even need to associate bucket with a service account for using it or i can just ahead and create a bucket and use it for connection?

Once I am able to do all this manually and setup the connection - i will use Terraform and other tools to automate.

Appreciate help and suggestions!

**Comment by drakkan** [comment]
Hi,

check the SFTPGo logs to understand the issue and take a look to the docs [here](https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application) to setup the authentication. If you want to use automatic credentials please note that the environment variable must be available for the user running SFTPGo

**Comment by drakkan** [comment]
> Hi drakkan,
> 
> I am trying to connect to my gcp bucket from Cyberduck through a user created on sftp/go. Steps followed are:
> 
> 1. Provisioned a GKE cluster with a service account and required permissions.
> 2. Created a bucket (not sure if i have to give that service account access to this bucket as well - for now it is not there)
> 3. Deployed sftp/go docker image on my project's container registry and was able to deploy and expose it on the container.
> 4. Now the sftp/go UI came up and i was able to create an admin user and then a sftpgo user from admin portal.
> 5. Choose the Google cloud storage as backend and checked the automatic credentials box.
> 6. Also added my public key for this user.
> 7. Now when i go to cyberduck and fill in the ip , 2022 as port , username and select my private key file - it is not able to make a connection to the server and upload to bucket.
> 
> Some points: i have not provisioned external DB - doing this for learning purpose so using inbuilt sqlite in sftpgo Running on 1 pod only to retain data Do i even need to associate bucket with a service account for using it or i can just ahead and create a bucket and use it for connection?
> 
> Once I am able to do all this manually and setup the connection - i will use Terraform and other tools to automate.
> 
> Appreciate help and suggestions!

This [article](https://medium.com/google-cloud/sftpgo-access-to-gcs-via-sftp-e203e0783f6f) answers all your questions
